GitUI/Editor/BlameAuthorMargin.cs

```diff
@@ -0,0 +1,110 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using System.Windows.Forms;
+using GitExtUtils.GitUI;
+using ICSharpCode.TextEditor;
+
+namespace GitUI.Editor
+{
+    /// <summary>
+    /// This class display avatars in the gutter in a blame control.
+    /// </summary>
+    public class BlameAuthorMargin : AbstractMargin
+    {
+        private static readonly int RecencyIndexWidth = Convert.ToInt32(4 * DpiUtil.ScaleX);
+        private List<Image> _avatars;
+        private readonly int _lineHeight;
+        private readonly Color _backgroundColor;
+        private List<GitBlameEntry> _blameLines;
+        private readonly Dictionary<int, SolidBrush> _brushs = new Dictionary<int, SolidBrush>();
+        private bool _isVisible = true;
+
+        public BlameAuthorMargin(TextArea textArea) : base(textArea)
+        {
+            _lineHeight = GetFontHeight(textArea.Font);
+            _backgroundColor = SystemColors.Window;
+            Width = _lineHeight + RecencyIndexWidth + DpiUtil.Scale(2);
+        }
+
+        public override int Width { get; }
+        public override bool IsVisible => _isVisible;
+
+        private static int GetFontHeight(Font font)
+        {
+            var max = Math.Max(
+                TextRenderer.MeasureText("_", font).Height,
+                (int)Math.Ceiling(font.GetHeight()));
+
+            return max + 1;
+        }
+
+        public void Initialize(List<GitBlameEntry> blameLines)
```

Comment:
❗️ Passing mutable collections can lead to hard-to-track bugs.
Since we're re-enumerating and re-materialising the collection few lines down, pass `IEnumerable<>`. 
The same applies to upstream calls.